clients/src/test/java/org/apache/kafka/test/TestSslUtils.java

```diff
@@ -399,10 +399,12 @@ public CertificateBuilder(int days, String algorithm) {
         }
 
         public CertificateBuilder sanDnsNames(String... hostNames) throws IOException {
-            GeneralName[] altNames = new GeneralName[hostNames.length];
-            for (int i = 0; i < hostNames.length; i++)
-                altNames[i] = new GeneralName(GeneralName.dNSName, hostNames[i]);
-            subjectAltName = GeneralNames.getInstance(new DERSequence(altNames)).getEncoded();
+            if (hostNames.length > 0) {
+                GeneralName[] altNames = new GeneralName[hostNames.length];
+                for (int i = 0; i < hostNames.length; i++)
+                    altNames[i] = new GeneralName(GeneralName.dNSName, hostNames[i]);
+                subjectAltName = GeneralNames.getInstance(new DERSequence(altNames)).getEncoded();
+            }
```

Comment:
Should we also add an `else` clause that sets `subjectAltName = null` to preserve the override semantics if the method is called twice? Just in case some test relies on overriding existing hostname in future.

Reply:
Makes sense. Added the `else` clause. Thank you!